compiler/next/include/chpl/resolution/scope-types.h

```diff
@@ -278,12 +278,38 @@ class Scope {
   }
 
   void stringify(std::ostream& ss, chpl::StringifyKind stringKind) const {
-    ss << "Scope ";
-    ss << tagToString(tag());
-    ss << " ";
+
+    // Print out header information (type of AST node, num entries).
+    ss << tagToString(tag()) << " '";
+    name_.stringify(ss, stringKind);
+    ss << "' (id: " ;
     id().stringify(ss, stringKind);
-    ss << " ";
-    ss << std::to_string(numDeclared());
+    ss << ", " << std::to_string(numDeclared()) << " entries)";
+    ss << std::endl;
+
+    // Find the largest entry name to pretty print...
+    int largestEntrySize = 0;
+    for (const auto& entry : declared_) {
+      int entryLength = entry.first.length();
+      largestEntrySize = entryLength > largestEntrySize
+          ? entryLength
+          : largestEntrySize;
+    }
+
+    // Print out each entry.
+    for (const auto& entry : declared_) {
+      auto borrowedIds = BorrowedIdsWithName(entry.second);
+      ss << "  ";
+      entry.first.stringify(ss, stringKind);
+
+      // Add spaces to align columns.
+      const int entryLength = entry.first.length();
+      const int numSpaces = largestEntrySize - entryLength;
+      for (int i = 0; i < numSpaces; i++) ss << " ";
+
+      ss << " : " << std::to_string(borrowedIds.numIds()) << " ids";
+      ss << std::endl;
+    }
```

Comment:
This is great but it is long enough that it shouldn't be in the .h anymore. Can you move it to scope-types.cpp?